src/components/sharing/index.js

```diff
@@ -220,7 +196,17 @@ function Sharing(props) {
                         }
                     });
                 });
-                if (failures.length > 0) {
+                if (failures.length === 1) {
+                    const { error_code: code, ...rest } = failures[0].error;
+                    setErrorDetails({
+                        message: tSharing("sharingUnsuccessful"),
+                        error: {
+                            response: {
+                                data: { error_code: code, reason: rest },
+                            },
+                        },
+                    });
+                } else if (failures.length > 0) {
                     setErrorDetails({
                         message: tSharing("sharingUnsuccessful"),
                         error: failures,
```

Comment:
May as well fix the error reporting here too?
```suggestion
                        error: { response: { data: failures } },
```
Might be kind of ugly for now, but it is only for the error details dialog, and I think a JSON string that users can report to support / intercom is better than checking the logs? 🤷

Reply:
Hm, I think that wouldn't do anything as-is, it'd just produce a mostly-empty error dialog (or possibly an error, since it might try to use the array as an object? not sure). I guess I could dump everything into `response.data.reason`, though, at least. I honestly don't have any idea how often there's more than one failure at this stage anyway, but you're right it's at least worth showing _something_

Reply:
Oops, that's right, I keep forgetting that it would have to be set inside `response.data.reason` or `response.data.message` 👍 

The only time I think there would be more than 1 failure is if sharing with multiple users, and more than 1 of those users had a sharing error.